<launch>
    <env name="LD_PRELOAD" value="/usr/lib/aarch64-linux-gnu/libgomp.so.1" />

    <arg name="output"              default="screen"                doc="display output to screen or log file"/>
    <arg name="odom_topic"          default="/SLAM/Pose"            doc="ROS topic for the pose"/>
    <arg name="control_topic"       default="/Control"              doc="ROS topic for control input"/>
    <arg name="obstacle_topic"      default="/Prediction/Obstacles" doc="ROS topic for obstacles"/>
    <arg name="traj_topic"          default="/Planning/Trajectory"  doc="ROS topic for trajectory"/>
    <arg name="path_topic"          default="/Routing/Path"         doc="ROS topic for reference path"/>
    <arg name="receding_horizon"    default="false"                 doc="whether to use open loop planner"/>
    <arg name="replan_dt"           default="0.1"                   doc="time step for replanning"/>
    <arg name="ilqr_config"         default="configs/lab1.yaml"     doc="path to ilqr config file"/>

    <node pkg="racecar_planner" type="traj_planning_node.py" name="traj_planning" output="$(arg output)">
        <param name="odom_topic"           value="$(arg odom_topic)" />
        <param name="control_topic"        value="$(arg control_topic)" />
        <param name="obstacle_topic"       value="$(arg obstacle_topic)" />
        <param name="traj_topic"           value="$(arg traj_topic)" />
        <param name="path_topic"           value="$(arg path_topic)" />
        <param name="package_path"         value="$(find racecar_planner)" />
        <param name="simulation"           value="False" />
        <param name="replan_dt"            value="$(arg replan_dt)" />
        <param name="receding_horizon"            value="$(arg receding_horizon)" />
        <param name="ilqr_params_file"     value="$(arg ilqr_config)" />
        <param name="PWM_model"            value="$(find racecar_planner)/configs/mlp_model.sav" />
    </node>

</launch>
